### PR TITLE
Allow fees to assigned by/to a variable price (price_id)

### PR DESCRIPTION
**Pull request by mintplugins**
Currently, fees are assigned to the download id but not the variable price. This means that fees get assigned multiple times for items in the cart with variable prices and fees.

This makes it so that a fee knows which price_id it belongs to - and only gets added to the cart_details if it (the fee) matches the price_id to which it belongs.

### CODE REVIEW

**Comment by mintplugins** [comment]
Issue 4477: https://github.com/easydigitaldownloads/easy-digital-downloads/issues/4477

File: includes/cart/functions.php [hunk 1]
```diff
@@ -58,7 +58,7 @@ function edd_get_cart_content_details() {
 		$discount   = edd_get_cart_item_discount_amount( $item );
 		$discount   = apply_filters( 'edd_get_cart_content_details_item_discount_amount', $discount, $item );
 		$quantity   = edd_get_cart_item_quantity( $item['id'], $item['options'] );
-		$fees       = edd_get_cart_fees( 'fee', $item['id'] );
+		$fees       = edd_get_cart_fees( 'fee', $item['id'], isset( $item['options']['price_id'] ) ? $item['options']['price_id'] : NULL );
```
Comment (pippinsplugins) [inline includes/cart/functions.php]: Let's get price_id defined on its own line.

File: includes/class-edd-fees.php [hunk 2]
```diff
@@ -196,6 +199,22 @@ public function get_fees( $type = 'fee', $download_id = 0 ) {
 			}
 
 		}
+		
+		//Now that we've removed any fees that are for other Downloads, lets also remove any fees that don't match this price id (or Variable Price)
+		if( ! empty( $fees ) && ! empty( $download_id ) && ! empty( $price_id ) ) {
```
Comment (pippinsplugins) [inline includes/class-edd-fees.php]: For price ID, we'll need to explicitly check that it is an `int` that is `0` or greater. Passing in `0`, which is a valid ID for some cases, would not pass the `! empty()` check.

**Comment by mintplugins** [comment]
@pippinsplugins I made those changes you mentioned above

**Comment by mintplugins** [comment]
Oh hold on...That empty check needs to pass if it is number - not fail

**Comment by pippinsplugins** [comment]
@mintplugins Could you update the unit tests for this too so they pass? https://travis-ci.org/easydigitaldownloads/easy-digital-downloads/jobs/124021824

If you're unsure how, ping me :)

**Comment by mintplugins** [comment]
I'm just committing tests to test the tests because I can't get tests to work locally.

**Comment by pippinsplugins** [comment]
Opening new PR against `release/2.6`.

**Pull request closed without merging**